easypost/util.py

```diff
@@ -0,0 +1,36 @@
+from typing import List
+
+from easypost.easypost_object import EasyPostObject
+from easypost.error import Error
+
+
+class Util:
+    @staticmethod
+    def _get_lowest_object_rate(
+        easypost_object: EasyPostObject,
+        carriers: List[str] = None,
+        services: List[str] = None,
+        rates_key: str = "rates",
+    ):
+        """Gets the lowest rate of an EasyPost object such as a Shipment, Order, or Pickup."""
+        carriers = carriers or []
+        services = services or []
+        lowest_rate = None
+
+        carriers = [carrier.lower() for carrier in carriers]
+        services = [service.lower() for service in services]
+
+        for rate in easypost_object[rates_key]:
```

Comment:
`easypost_object[rates_key]` could produce `None` if it doesn't exist (rates haven't been loaded in the object yet), which will produce an error. Try `easypost_object.get(rates_key, [])` to default to an empty list if `rates_key` doesn't exist.

Reply:
Although true, as a point of clarification, a rates key should always exist on these object. But this is a good best practice regardless.